cmd/dockerregistry/main.go

```diff
@@ -16,6 +18,18 @@ import (
 )
 
 func main() {
+	// Start the profiling server, if requested.
+	profilingAddr := os.Getenv("REGISTRY_PROFILING_ADDR")
```

Comment:
We should use that anyway because it handles panics and let's us do CPU
profiling of an entire run.

Crib the same code that is in other binaries for running pprof web
(including the extra env var for addr).  That code should go into
serviceability but you don't have to move it in this PR.

On Mar 4, 2016, at 9:48 AM, Dan Mace <notifications@github.com> wrote:

In cmd/dockerregistry/main.go
<https://github.com/openshift/origin/pull/7771#discussion_r55038046>:

> @@ -16,6 +18,18 @@ import (
>  )
>
>  func main() {
> +	// Start the profiling server, if requested.
> +	profilingAddr := os.Getenv("REGISTRY_PROFILING_ADDR")

@smarterclayton <https://github.com/smarterclayton>

Use the same env we use on Openshift binaries, both address and the web
bit. In fact, just use the serviceability package.

I was just looking through the serviceability package, and all it provides
is a deferred shutdown hook which writes a single kind of profiler output
to a file on program exit. That's not equivalent to what I'm providing in
this PR (a live net/pprof session over HTTP). Am I missing something?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/7771/files#r55038046>.